src/utils/cairoParsing.ts

```diff
@@ -0,0 +1,93 @@
+import assert from 'assert';
+import { TranspileFailedError } from './errors';
+import { Implicits, implicitTypes } from './implicits';
+
+export type RawCairoFunctionInfo = {
+  name: string;
+  implicits: Implicits[];
+};
+
+/**
+ * Given several Cairo function represented in plain text extracts information from it
+ *  @param rawFunctions Multiple cairo functions in a single text
+ *  @returns A list of each function information
+ */
+export function parseMultipleRawCairoFunctions(rawFunctions: string): RawCairoFunctionInfo[] {
+  const functions = rawFunctions.matchAll(/func (\w+)\s*[{]?.*?[}]?\s*[(].*?[)]/gis);
+
+  return [...functions].map((func) => getRawCairoFunctionInfo(func[0]));
+}
```

Comment:
What if there is a comment with `{`, `}`, `(`, or `)` within cairo stub? Should we consider such an edge case? If we want to support any code compatible with cairo syntax we have to use a parser instead of regex

Reply:
If the `{}()` symbols are inside the function signature, then it will crash. If they are in the body definition, then they will be ignored (unless the comment includes a proper func signature like `func name{...}(...)`) So yes i definitely agree to the use of a parser because an adversarial user could break the regex easily. I would propose making a task for it once the Cairo1 transition is complete. (I think a parser, and cairo-ast in typescript can be helpful to the transpiler)

Reply:
Sure, we should definitely make an issue for that

Reply:
Here #949